packages/core/src/baseclient.ts

```diff
@@ -628,13 +630,14 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
     }
 
     const isTransaction = event.type === 'transaction';
-    const beforeSendProcessorName = isTransaction ? 'beforeSendTransaction' : 'beforeSend';
-    const beforeSendProcessor = options[beforeSendProcessorName];
+    const isError = !event.type;
```

Comment:
The alternative implementation would be IMHO:

```js
if (!isTransaction && !isError) {
  // skip...
}
```

This would probably by smaller in footprint, but we may need to revisit this if/when we also want these hooks for e.g. replay or profiles.

Reply:
hmm probably missing something but why not use the new `isErrorEvent` function?

Reply:
good point!